src/flamegraph/mod.rs

```diff
@@ -557,21 +503,28 @@ where
         let frame_attributes = opt
             .func_frameattrs
             .frameattrs_for_func(frame.location.function);
-        let frame_attributes = override_or_add_attributes(&buffer[info], frame_attributes);
-        let href_is_some = frame_attributes.href.is_some();
 
-        if href_is_some {
-            write_container_attributes(&mut cache_a, &frame_attributes);
-            svg.write_event(&cache_a)?;
-        } else {
-            write_container_attributes(&mut cache_g, &frame_attributes);
+        let mut has_href = false;
+        let mut title = &buffer[info];
+        if let Some(frame_attributes) = frame_attributes {
+            if frame_attributes.attrs.contains_key("xlink:href") {
+                write_container_attributes(&mut cache_a, &frame_attributes);
+                svg.write_event(&cache_a)?;
+                has_href = true;
+            } else {
+                write_container_attributes(&mut cache_g, &frame_attributes);
+                svg.write_event(&cache_g)?;
+            }
+            if let Some(ref t) = frame_attributes.title {
+                title = t.as_str();
+            }
+        } else if let Event::Start(ref mut c) = cache_g {
+            c.clear_attributes();
```

Comment:
Why does this need to clear the cache when the other arms above don't?

Reply:
The other arms do in `write_container_attributes`, but I could pull it out and put it above the `if`.

Reply:
Actually, I guess it can't go above the `if` because `c` doesn't exist there.

Reply:
Ohh, I missed that it was done in the function. Seems fine then :+1: